catalog/core/catalog-core-standardframework/src/test/java/ddf/catalog/history/HistorianTest.java

```diff
@@ -0,0 +1,602 @@
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Callable;
+
+import org.apache.shiro.subject.ExecutionException;
+import org.codice.ddf.security.common.Security;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+
+import ddf.catalog.content.StorageException;
+import ddf.catalog.content.StorageProvider;
+import ddf.catalog.content.data.ContentItem;
+import ddf.catalog.content.operation.CreateStorageResponse;
+import ddf.catalog.content.operation.DeleteStorageRequest;
+import ddf.catalog.content.operation.ReadStorageRequest;
+import ddf.catalog.content.operation.ReadStorageResponse;
+import ddf.catalog.content.operation.StorageRequest;
+import ddf.catalog.content.operation.UpdateStorageRequest;
+import ddf.catalog.content.operation.UpdateStorageResponse;
+import ddf.catalog.content.operation.impl.DeleteStorageRequestImpl;
+import ddf.catalog.core.versioning.DeletedMetacard;
+import ddf.catalog.core.versioning.MetacardVersion;
+import ddf.catalog.core.versioning.impl.DeletedMetacardImpl;
+import ddf.catalog.core.versioning.impl.MetacardVersionImpl;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.data.impl.BasicTypes;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.filter.proxy.builder.GeotoolsFilterBuilder;
+import ddf.catalog.operation.CreateRequest;
+import ddf.catalog.operation.DeleteRequest;
+import ddf.catalog.operation.DeleteResponse;
+import ddf.catalog.operation.SourceResponse;
+import ddf.catalog.operation.Update;
+import ddf.catalog.operation.UpdateRequest;
+import ddf.catalog.operation.UpdateResponse;
+import ddf.catalog.operation.impl.DeleteRequestImpl;
+import ddf.catalog.operation.impl.DeleteResponseImpl;
+import ddf.catalog.operation.impl.UpdateImpl;
+import ddf.catalog.source.CatalogProvider;
+import ddf.catalog.source.IngestException;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.catalog.source.UnsupportedQueryException;
+import ddf.security.Subject;
+
+public class HistorianTest {
+
+    private static final String METACARD_ID = "METACARD_ID";
+
+    private static final String DELETED_METACARD_ID = "DELETED_ID";
+
+    private static final String VERSIONED_METACARD_ID = "VERSIONED_ID";
+
+    private static final String RESOURCE_URI = "content:example.com";
+
+    private static final String UPDATE_DESCRIPTION = "This is an updated description.";
+
+    private CatalogProvider catalogProvider;
+
+    private InMemoryStorageProvider storageProvider;
+
+    private Historian historian;
+
+    @Before
+    public void setup() {
+        historian = new Historian();
+
+        catalogProvider = mock(CatalogProvider.class);
+        historian.setCatalogProviders(Collections.singletonList(catalogProvider));
+
+        storageProvider = new InMemoryStorageProvider();
```

Comment:
It was a lot easier to assert on the adding/removing of objects in this than trying to get a mock to use its own map.